src/erlcloud_aws.erl

```diff
@@ -974,13 +976,14 @@ timestamp_to_gregorian_seconds(Timestamp) ->
 get_credentials_from_metadata(Config) ->
     %% TODO this function should retry on errors getting credentials
     %% First get the list of roles
-    case erlcloud_ec2_meta:get_instance_metadata("iam/security-credentials/", Config) of
+    IMDSv2Token = maybe_imdsv2_session_token(Config),
+    case erlcloud_ec2_meta:get_instance_metadata("iam/security-credentials/", Config, IMDSv2Token) of
```

Comment:
maybe I made a stylish mistake with previous PR and would prefer `Config` to always be the last parameter, but I guess it's late